frontend/packages/console-shared/src/hooks/screenfull.ts

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import * as screenfull from 'screenfull';
+
+export const useFullScreen = (node: HTMLElement): [boolean, () => void] => {
+  const [isFullScreen, setIsFullScreen] = React.useState<boolean>(false);
+
+  const fullScreenToggleCallback = React.useCallback(() => {
+    if (node && screenfull.enabled) {
+      screenfull.toggle(node);
+    }
+  }, [node]);
+
+  React.useEffect(() => {
+    if (screenfull.enabled) {
+      screenfull.on('change', () => {
+        setIsFullScreen(screenfull.isFullscreen);
```

Comment:
As a generic component you cannot assume that a global fullscreen event relates to this particular instance of the hook. You need to associate the event back to the node associated with this hook instance.